BJS2-15552

shade2903 opened the pull request:
Create RecommendationDto;
Create SkillOfferDto;

shade2903 committed:
BJS2-15552

Create RecommendationDto;
Create SkillOfferDto;

shade2903 committed:
BJS2-15552

create class DataValidationException;
create class RecommendationController with methods:
-RecommendationDto giveRecommendation(RecommendationDto recommendation);
-RecommendationDto updateRecommendation(RecommendationDto updated);
-deleteRecommendation(long id);
-List<RecommendationDto> getAllUserRecommendations(long receiverId, Pageable pageable)
-List<RecommendationDto> getAllGivenRecommendations(long authorId, Pageable pageable)

create class RecommendationValidator implementing service validation;
create class RecommendationService implementing controllers methods;

shade2903 committed:
BJS2-15552

create RecommendationControllerTest
create RecommendationValidatorTest

shade2903 committed:
BJS2-15552

create Service RecommendationServiceTest

shade2903 committed:
BJS2-15552

create Service RecommendationServiceTest

shade2903 committed:
BJS2-15552

create Service RecommendationServiceTest

ravshanbekn commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,131 @@
+    public List<RecommendationDto> getAllUserRecommendations(long receiverId) {
+        recommendationValidator.validateId(receiverId);
+        List<Recommendation> allUserRecommendation = recommendationRepository.findAllByReceiverId(receiverId);
+        if (allUserRecommendation != null || !allUserRecommendation.isEmpty()) {
```
Тут случаем не &&?
Если allUserRecommendation != null, но empty, то из-за того, что || if сработает

shade2903 replied:
Спасибо,что заметил, ты мне баг помог отловить  !allUserRecommendation.isEmpty() тут вообще лишнее, я дальше вызываю метод валидации который провереяет на isEmpty(); получается два раза проверил.

ravshanbekn commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,131 @@
+    public List<RecommendationDto> getAllGivenRecommendations(long authorId) {
+        recommendationValidator.validateId(authorId);
+        List<Recommendation> allUserRecommendation = recommendationRepository.findAllByReceiverId(authorId);
+        if (allUserRecommendation != null || !allUserRecommendation.isEmpty()) {
```
По моему, тут тоже

shade2903 replied:
И здесь баг отловил.

ravshanbekn commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,131 @@
+    public void saveGuaranteeUserSkill(RecommendationDto recommendationDto) {
+        Recommendation recommendation = recommendationMapper.toEntity(recommendationDto);
+        List<Skill> userSkills = skillRepository.findAllByUserId(recommendation.getReceiver().getId());
+        if (recommendation.getSkillOffers() != null || !recommendation.getSkillOffers().isEmpty()) {
```
Тоже

shade2903 replied:
поправил

ravshanbekn commented on src/test/java/school/faang/user_service/service/RecommendationServiceTest.java:
```diff
@@ -0,0 +1,175 @@
+                .skillOffers(skillOfferDtoList)
+                .build();
+        Recommendation recommendation =
+                Recommendation.builder()
```
Этот код повторяется может в beforeEach его?
edited: криво выделяю сори, с 85 до 106 строки

shade2903 replied:
Спасибо. Поправил.

ravshanbekn commented on src/test/java/school/faang/user_service/service/RecommendationServiceTest.java:
```diff
@@ -0,0 +1,175 @@
+                RecommendationDto.builder()
+                        .id(recommendationId)
+                        .authorId(receiverId)
+                        .build());
```
Тоже с предыдущим тестом повторяется, в beforeEach может?

shade2903 replied:
Тоже поправил.

ravshanbekn reviewed (commented):
В целом, гуд)

shade2903 committed:
BJS2-15552

fix bags in RecommendationService;
add method init() in RecommendationServiceTest

shade2903 committed:
BJS2-15552

add method init() in RecommendationControllerTest

pablo023 commented:
Пофикси конфликты

shade2903 committed:
BJS2-15552

conflict resolve in DataValidationException

shade2903 committed:
BJS2-15552

conflict resolve in DataValidationException

shade2903 committed:
BJS2-15552

conflict resolve in DataValidationException

shade2903 committed:
Merge branch 'titan-master-bc5' into feature-BJS2-15552

shade2903 commented:
пофиксил конфликт

shade2903 committed:
BJS2-15552

fix bags RecommendationValidatorTest in methods:
-validateNotValidIdTest();
-validateRecommendationAuthorIdTest();
-validateRecommendationReceiverIdTest()
